homeassistant/components/schluter/__init__.py

```diff
@@ -0,0 +1,86 @@
+"""The Schluter DITRA-HEAT integration."""
+from datetime import timedelta
+import logging
+
+from requests import RequestException, Session
+from schluter.api import Api
+from schluter.authenticator import AuthenticationState, Authenticator
+import voluptuous as vol
+
+from homeassistant.const import CONF_PASSWORD, CONF_TIMEOUT, CONF_USERNAME
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+DATA_SCHLUTER_SESSION = "schluter_session"
+DATA_SCHLUTER_API = "schluter_api"
+PLATFORMS = ["climate"]
+DEFAULT_TIMEOUT = 10
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=5)
+SCHLUTER_CONFIG_FILE = ".schluter.conf"
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_USERNAME): cv.string,
+                vol.Required(CONF_PASSWORD): cv.string,
+                vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup_schluter(hass, config, api, authenticator):
+    """Set up the Schluter component."""
+
+    authentication = None
+    try:
+        authentication = authenticator.authenticate()
+    except RequestException as ex:
+        _LOGGER.error("Unable to connect to Schluter service: %s", ex)
+
+    state = authentication.state
+
+    if state == AuthenticationState.AUTHENTICATED:
+        hass.data[DATA_SCHLUTER_SESSION] = authentication.session_id
+        hass.data[DATA_SCHLUTER_API] = api
```

Comment:
`hass.data` must be namespaced by the domain
```suggestion
        hass.data[DOMAIN] = {
            DATA_SCHLUTER_API: api,
            DATA_SCHLUTER_SESSION: authentication.session_id,
        }
```

Reply:
Thanks for the suggested change, I wasn't quite sure I was doing that properly.